code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -35,6 +35,7 @@
 	var/weaponscheck = FALSE //If true, arrest people for weapons if they lack access
 	var/check_records = TRUE //Does it check security records?
 	var/arrest_type = FALSE //If true, don't handcuff
+	var/ranged = FALSE //used for EDs

```

Comment:
This is an istype(src) check in disguise.

Reply:
is that fine or do i have to replace with istype

Reply:
you use inheritance